### ISSUE TITLE
Вложенные селекторы и изменение блока контекстом

### ISSUE THREAD

**Issue by ghost** [bem-site/bem-forum-content-ru#1603]
Пример разметки:
```
.product
   .rate.product__rate
        .rate__stars
```

Пример css-правила:
```
.product__rate .rate__stars { /*...*/ }
```
Я правильно понимаю, стилевое правило выше противоречит рекомендациям, описанных во [Вложенные селекторы](https://ru.bem.info/methodology/css/#%D0%B2%D0%BB%D0%BE%D0%B6%D0%B5%D0%BD%D0%BD%D1%8B%D0%B5-%D1%81%D0%B5%D0%BB%D0%B5%D0%BA%D1%82%D0%BE%D1%80%D1%8B)?
Какие есть варианты модификации `.rate__stars` кроме использования модификатора на этом элементе или его родителе?

Читая документацию, обратил внимание на следующий пункт:  
_Изменение блока производится при помощи размещения одного блока в составе другого. Правила родительского блока применяются каскадом к вложенным блокам._
[Изменение блока контекстом](https://ru.bem.info/methodology/block-modification/#%D0%B8%D0%B7%D0%BC%D0%B5%D0%BD%D0%B5%D0%BD%D0%B8%D0%B5-%D0%B1%D0%BB%D0%BE%D0%BA%D0%B0-%D0%BA%D0%BE%D0%BD%D1%82%D0%B5%D0%BA%D1%81%D1%82%D0%BE%D0%BC)

Эта рекомендация относится исключительно к правилам, наследуемых по каскаду от родителя? Вложенный селектор в примере сбил с толку, невольно появились параллели с примером `.product__rate .rate__stars`.

**Comment by Realetive** [comment]
Без контекста правильный совет дать сложно, но предположу, что блок `.rate` используется с разным оформлением, т. е. логично ввести модификатор:

```
.product
   .rate.rate_view_product.product__rate
        .rate__stars
```

```
.rate_view_product .rate__stars {}
```

В противном случае (если rate везде одинаковый), стили для rate__stars пишутся в самом .rate__stars.

**Comment by ghost** [comment]
> предположу, что блок .rate используется с разным оформлением

Да, спасибо.

По вопросу, связанным с [Изменение блока контекстом](https://ru.bem.info/methodology/block-modification/#%D0%B8%D0%B7%D0%BC%D0%B5%D0%BD%D0%B5%D0%BD%D0%B8%D0%B5-%D0%B1%D0%BB%D0%BE%D0%BA%D0%B0-%D0%BA%D0%BE%D0%BD%D1%82%D0%B5%D0%BA%D1%81%D1%82%D0%BE%D0%BC) - можете привести пример использования на практике или собственное объяснение "что имеется в виду"? Вроде с виду все просто, а мозг почему-то завис в непонимании.

**Comment by Realetive** [comment]
Да. Например, у нас есть какой-то блок, на разметку которого мы не можем влиять CSS-классами — WYSIWIG-редактор, который заполняют менеджеры в админке (или markdown-разметка). Тут каскад — единственный «дешёвый» способ «достучаться» до оформления вложенных блоков.
